shared/config.go

```diff
@@ -305,18 +307,57 @@ func WithIdleHTTPConnectionTimeout(value time.Duration) Option {
 	}
 }
 
-// WithHTTPTransportWrapper provides ability to control http transport
-// For testing purposes only, don't use it on production
-func WithHTTPTransportWrapper(wrapper func(http.RoundTripper) http.RoundTripper) Option {
+// GzipCompressionFunc creates a gzip compression function with the specified compression level.
+// Level can be:
+//   - gzip.HuffmanOnly (-2): Huffman compression only
+//   - gzip.DefaultCompression (-1): default compression level (recommended)
+//   - gzip.NoCompression (0): no compression (store only)
+//   - gzip.BestSpeed (1): fastest compression
+//   - gzip.BestCompression (9): best compression ratio
+func GzipCompressionFunc(level int) RequestCompressionFunc {
```

Comment:
Could you please make a configuration struct for it `GzipConfig`, with public setter WithLevel(level int) and a constructor `NewGzipConfig` and make level default to -1.
Later we will add minimal body len limit for it and also make it return chunks after certain body size.
Also lets rename it to `GzipRequestCompressor`: